src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleKeyCache.scala

```diff
@@ -73,11 +73,17 @@ class GoogleKeyCache(val googleIamDAO: GoogleIamDAO, val googleStorageDAO: Googl
     keyFuture.value.map(_.getOrElse(throw new WorkbenchException("Unable to furnish new key")))
   }
 
-  private def retrieveActiveKey(pet: PetServiceAccount, keyObjects: List[GcsObjectName]): Future[String] = {
+  private def retrieveActiveKey(pet: PetServiceAccount, keyObjects: List[GcsObjectName], serviceAccountKeys: List[ServiceAccountKey]): Future[String] = {
     val mostRecentKey = keyObjects.sortBy(_.timeCreated.toEpochMilli).last
     val keyRetired = System.currentTimeMillis() - mostRecentKey.timeCreated.toEpochMilli > 86400000L * googleServicesConfig.googleKeyCacheConfig.activeKeyMaxAge
 
-    if(keyRetired) furnishNewKey(pet)
+    val keyPathPattern = """([^\/]+)\/([^\/]+)\/([^\/]+)""".r
+    val keyPathPattern(project, petSaEmail, keyId) = mostRecentKey.value
+
+    //The key may exist in the Google bucket cache, but could have been deleted from the SA directly
+    val keyExistsForSA = serviceAccountKeys.exists(_.id.value.contentEquals(keyId))
+
+    if(keyRetired || !keyExistsForSA) furnishNewKey(pet)
```

Comment:
also we do not need to worry about deleting the key, we can wait for the lifecycle to pick them up

Reply:
what happens if key is already gone from google when deleted from bucket?

Reply:
it's handled without an issue:

1) pub/sub message is triggered
2) sam tries to delete it, gets a 404 which recovers gracefully in HttpGoogleIamDAO 
3) sam sees this as a success and acks the message